c/zis/services/auth.c

```diff
@@ -66,74 +66,77 @@ static int handleVerifyPassword(AuthServiceParmList *parmList,
-      int rc = RC_ZIS_AUTHSRV_OK;
-      IDTA *idta = NULL;
-      int options = VERIFY_CREATE;
-
-      if (parmList->options & ZIS_AUTH_SERVICE_PARMLIST_OPTION_GENERATE_IDT) {
-        idta = (IDTA *) safeMalloc31(sizeof(IDTA), "Idta structure");
-        memset(idta, 0, sizeof(IDTA));
-        memcpy(idta->id, "IDTA", 4);
-        idta->version = IDTA_VERSION_0001;
-        idta->length = sizeof(IDTA);
-        idta->idtType = IDTA_JWT_IDT_Type;
-        idta->idtBufferPtr = parmList->safIdtService.safIdt;
-        idta->idtBufferLen = sizeof(parmList->safIdtService.safIdt);
-        idta->idtLen = parmList->safIdtService.safIdtLen;
-        idta->idtPropIn = IDTA_End_User_IDT;
-        options |= VERIFY_GENERATE_IDT;
-      }
-
-      CMS_DEBUG(globalArea, "handleGenerateToken(): username = %s, password = %s\n",
-          parmList->userIDNullTerm, "******");
-        if (parmList->options & ZIS_AUTH_SERVICE_PARMLIST_OPTION_IDT_APPL) {
-          safRC = safVerify7(options, parmList->userIDNullTerm,
-            parmList->passwordNullTerm, &acee, parmList->safIdtService.applNullTerm, &racfRC, &racfRsn, idta);
-        } else {
-          safRC = safVerify6(options, parmList->userIDNullTerm,
-            parmList->passwordNullTerm, &acee, &racfRC, &racfRsn, idta);
-        }
-
-      CMS_DEBUG(globalArea, "safVerify(VERIFY_CREATE) safStatus = %d, RACF RC = %d, "
-          "RSN = %d, ACEE=0x%p\n", safRC, racfRC, racfRsn, acee);
-      if (idta != NULL) {
-        CMS_DEBUG(globalArea, "IDTA token: gen_rc = %d, prop_out = %X, prop_in = %X "
-          "token length = %d\n", idta->idtGenRc, idta->idtPropOut, idta->idtPropIn,
-          idta->idtLen);
-      }
-
-      if (safRC != 0) {
-        rc = RC_ZIS_AUTHSRV_SAF_ERROR;
-        goto acee_deleted;
-      }
-
-      if (idta != NULL) {
-        parmList->safIdtService.safIdtLen = idta->idtLen;
-      }
-
-      deleteSAFRC = safVerify(VERIFY_DELETE, NULL, NULL, &acee, &deleteRACFRC,
-          &deleteRACFRsn);
-      CMS_DEBUG(globalArea, "safVerify(VERIFY_DELETE) safStatus = %d, RACF RC = %d, "
-          "RSN = %d, ACEE=0x%p\n", deleteSAFRC, deleteRACFRC, deleteRACFRsn,
-          acee);
-      if (deleteSAFRC != 0) {
-        rc = RC_ZIS_AUTHSRV_DELETE_FAILED;
-      }
-    acee_deleted:
-
-      FILL_SAF_STATUS(&parmList->safStatus, safRC, racfRC, racfRsn);
-      CMS_DEBUG(globalArea, "handleVerifyPassword() done\n");
-      if (idta != NULL) {
-        safeFree(idta, sizeof(IDTA));
-        idta = NULL;
-      }
-      return rc;
+   if(parmList->safIdtService.safIdtServiceVersion < SAF_IDT_SERVICE_CURRENT_VERSION) {
+        return RC_ZIS_AUTHSRV_BAD_SAF_SERVICE_VERSION;
+   }
+   ACEE *acee = NULL;
+   int safRC = 0, racfRC = 0, racfRsn = 0;
+   int deleteSAFRC = 0, deleteRACFRC = 0, deleteRACFRsn = 0;
+   int rc = RC_ZIS_AUTHSRV_OK;
+   IDTA *idta = NULL;
+   int options = VERIFY_CREATE;
+
+
+  idta = (IDTA *) safeMalloc31(sizeof(IDTA), "Idta structure");
+  if(idta == NULL) {
```

Comment:
Hmm, actually, can we just pass a stack variable? For example,
```c
IDTA idta = {
  .eyecatcher = "IDTA",
  .version = IDTA_VERSION_0001,
  .idtLen = parmList->safIdtService.safIdtLen,
  .options = VERIFY_GENERATE_IDT,
  .idtPropIn = IDTA_End_User_IDT,
};
. . .
safRC = safVerify6(options, parmList->userIDNullTerm, parmList->passwordNullTerm, &acee, &racfRC, &racfRsn, &idta);
```
And even if we don't use a stack variable, you can get rid of the NULL checks because you don't continue if `idta` is `NULL`.

Reply:
there are compilation errors when I tried this(ERROR CCN3022 ../../c/zis/services/auth.c:80    "eyecatcher" is not a member of "struct IDTA_tag".
ERROR CCN3022 ../../c/zis/services/auth.c:87    "options" is not a member of "struct IDTA_tag".) . Would it be ok if I just remove the null check then?

Reply:
That was just an example, the field name is probably different.